FreeRTOS/Test/CMock/message_buffer/message_buffer_utest.c

```diff
@@ -0,0 +1,739 @@
+    vTaskSuspendAll_Ignore();
+    xTaskResumeAll_IgnoreAndReturn( pdTRUE );
+
+    /* Create a message buffer of the default test sample size. */
+    xMessageBuffer = xMessageBufferCreate( TEST_MESSAGE_BUFFER_SIZE );
+    TEST_ASSERT_NOT_NULL( xMessageBuffer );
+    TEST_ASSERT_EQUAL( TEST_MESSAGE_BUFFER_SIZE, xMessageBufferSpacesAvailable( xMessageBuffer ) );
+
+    sent = xMessageBufferSend( xMessageBuffer, data, TEST_MAX_MESSAGE_SIZE, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, sent );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, xStreamBufferNextMessageLengthBytes( xMessageBuffer ) );
+
+    /* Receiving bytes of size less than the message length will return 0 bytes. */
+    received = xMessageBufferReceive( xMessageBuffer, dataReceived, TEST_MAX_MESSAGE_SIZE - 1, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    TEST_ASSERT_EQUAL( 0, received );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, xStreamBufferNextMessageLengthBytes( xMessageBuffer ) );
+
+    /* Receiving bytes of size greater than the message length will return message length bytes. */
+    received = xMessageBufferReceive( xMessageBuffer, dataReceived, TEST_MAX_MESSAGE_SIZE + 1, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    TEST_ASSERT_EQUAL( TEST_MAX_MESSAGE_SIZE, received );
+    TEST_ASSERT_EQUAL( 0, xStreamBufferNextMessageLengthBytes( xMessageBuffer ) );
+
+    vStreamBufferDelete( xMessageBuffer );
+}
+
+/**
+ * @brief Validates xMessageBufferReceive API with null input message.
+ */
+void test_xMessageBufferReceive_null_input_message( void )
+{
+    vTaskSetTimeOutState_Ignore();
+    vTaskSuspendAll_Ignore();
+    xTaskResumeAll_IgnoreAndReturn( pdTRUE );
+
+    /* Create a message buffer of sample size. */
+    xMessageBuffer = xMessageBufferCreate( TEST_MESSAGE_BUFFER_SIZE );
+    TEST_ASSERT_NOT_NULL( xMessageBuffer );
+
+    /* Should assert if a null input mssage is passed. */
+    if( TEST_PROTECT() )
+    {
+        ( void ) xMessageBufferReceive( xMessageBuffer, NULL, TEST_MAX_MESSAGE_SIZE, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    }
+
+    validate_and_clear_assertitions();
+    vStreamBufferDelete( xMessageBuffer );
+}
+
+
+/**
+ * @brief Validates xMessageBufferReceive API with null message buffer handle.
+ */
+void test_xMessageBufferReceive_invalid_params( void )
+{
+    uint8_t message[ TEST_MAX_MESSAGE_SIZE ] = { 0 };
+
+    vTaskSetTimeOutState_Ignore();
+    vTaskSuspendAll_Ignore();
+    xTaskResumeAll_IgnoreAndReturn( pdTRUE );
+
+    /* Create a message buffer of sample size. */
+    xMessageBuffer = xMessageBufferCreate( TEST_MESSAGE_BUFFER_SIZE );
+    TEST_ASSERT_NOT_NULL( xMessageBuffer );
+
+    /* Should assert if a null message buffer handle is passed. */
+    if( TEST_PROTECT() )
+    {
+        ( void ) xMessageBufferReceive( NULL, message, TEST_MAX_MESSAGE_SIZE, TEST_MESSAGE_BUFFER_WAIT_TICKS );
+    }
+
+    validate_and_clear_assertitions();
+
+    vStreamBufferDelete( xMessageBuffer );
```

Comment:
Similar issue here re: memory allocation / de-allocation.

Reply:
I think it's not the end of the world to leak a single xMessageBuffer though, especially since this is only running in a simulated environment.